CefSharp.OffScreen/ChromiumWebBrowser.cs

```diff
@@ -614,20 +678,77 @@ void IRenderWebBrowser.InvokeRenderAsync(BitmapInfo bitmapInfo)
         /// in it's lock scope.
         /// </summary>
         /// <param name="bitmapInfo">information about the bitmap to be rendered</param>
+        /// <exception cref="InvalidOperationException">When background image is null when on blend mode.</exception>
         public virtual void InvokeRenderAsync(BitmapInfo bitmapInfo)
         {
             var gdiBitmapInfo = (GdiBitmapInfo)bitmapInfo;
-            if (bitmapInfo.CreateNewBitmap)
+
+            if (BlendingConfiguration == PopupBlending.Separate)
             {
-                if (Bitmap != null)
+                if (bitmapInfo.CreateNewBitmap)
                 {
-                    Bitmap.Dispose();
-                    Bitmap = null;
+                    if (Bitmap != null)
+                    {
+                        Bitmap.Dispose();
+                        Bitmap = null;
+                    }
+
+                    Bitmap = gdiBitmapInfo.CreateBitmap();
                 }
+            }
+            else if (BlendingConfiguration == PopupBlending.Blend)
```

Comment:
What about moving this to `ScreenshotOrNull` and adding a `PopupBlending` as a param? Greatly simplify code, no need to store the `BackGroundImage`, just check if their is a popup open and merge on demand.

Performance would be a concern in your current implementation as every frame rendered would be `blended`, which is likely unnecessary, just the frames the user chooses to capture require blending.

Comments?